src/managers/quorum.py

```diff
@@ -228,8 +228,21 @@ def update_acls(self, event: RelationEvent | None = None) -> None:
             logger.debug(f"UPDATE CHROOT - {client.database}")
             self.client.set_acls_znode_leader(path=client.database, acls=[generated_acl])
 
+            subnodes = self.client.leader_znodes(path=client.database)
+            for node in subnodes:
+                self.client.set_acls_znode_leader(path=node, acls=[generated_acl])
+
         # Looks for applications no longer in the relation but still in config
+        restricted_acl = make_acl(
+            scheme="sasl",
+            credential="super",
+            read=True,
+            write=True,
+            create=True,
+            delete=True,
+            admin=True,
```

Comment:
**todo:** You can just pass `all=True` here

Reply:
Thanks, changed in 42876719eb74